bin/commands/indexDump.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable no-console */
+
+const
+  fs = require('fs'),
+  ColorOutput = require('./colorOutput'),
+  path = require('path'),
+  getSdk = require('./getSdk');
+
+function mkdirp (fullPath) {
+  const parts = fullPath.split(path.sep);
+
+  if (parts.length > 1) {
+    mkdirp(parts.slice(0, parts.length - 1).join('/'));
+  }
+
+  if (! fs.existsSync(fullPath)) {
+    fs.mkdirSync(fullPath);
+  }
+}
+
+function dumpCollectionPart (results, bulkDocuments) {
```

Comment:
This won't do: you're storing the entire collection in memory, but that means that we won't be able to dump even moderately large collections.

You need to use a [writable stream](https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_fs_createwritestream_path_options) and dump the result page per page, to keep memory consumption under control.